evalml/pipelines/components/transformers/preprocessing/drop_rows_transformer.py

```diff
@@ -13,6 +13,7 @@ class DropRowsTransformer(Transformer):
 
     name = "Drop Rows Transformer"
     modifies_target = True
+    training_only = True
```

Comment:
I've made this abstract, but `Transformer` still sets this value to False, so any subclass of `Transformer` will have a default value of False... which might defeat the purpose of having this as abstract 😅 

I *think* this is okay though, since most of our components will have `training_only` as False for now... but open to thoughts and concerns :)